### PR TITLE
[TOPI] Add layer norm operator

### PR DESCRIPTION
**Pull request by vinx13**
This PR added a tuple-sum based implementation of layer norm. It performs one-pass reduction to compute mean and variance at the same time.
Reducer pattern is also added to allow `LowerCrossThreadReduction` to handle this case.
On CUDA, it will generate two kernels: one for reduction and one for elemwise operations. Because of some limitation of `compute_at` currently we are not able to fuse them into one kernel. 

cc @MasterJH5574 @junrushao @AndrewZhaoLuo

### CODE REVIEW

File: include/tvm/topi/nn/layer_norm.h [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+using namespace tvm::te;
+
+/*!
+ * \brief Layer normalization.
+ * \param data N-D tensor with shape [d_0, d_1, ..., d_n]
+ * \param gamma R-D tensor with shape [r_0, r_1, ..., r_k] where R == len(axis) and d_{axis_i} ==
+ *              r_i
+ * \param beta Optional, R-D tensor with shape [r_0, r_1, ..., r_k] where R == len(axis) and
+ *             d_{axis_i} == r_i
```
Comment (MasterJH5574) [inline include/tvm/topi/nn/layer_norm.h:44]: For `data`, should the shape be `[d_0, ..., d_{n-1}]`? (Ditto for the documents on Python side)

**Review by MasterJH5574 — APPROVED** [review]
LGTM! Thanks @vinx13 for implementing layer-norm! I just have two tiny nits.

File: python/tvm/topi/nn/__init__.py [hunk 2]
```diff
@@ -38,6 +38,7 @@
 from .bnn import *
 from .qnn import *
 from .upsampling import *
+from .layer_norm import layer_norm
```
[the remark this answers is not in the document]
Comment (junrushao) [reply python/tvm/topi/nn/__init__.py:41]: Wildcard importing is actually not a good idea though lol

[the remark this answers is not in the document]
Comment (vinx13) [reply python/tvm/topi/nn/__init__.py:41]: agreed, so I avoid using wildcard here. perhaps we should clean up this file in the future

**Pull request merged**